File: java/src/main/java/io/fivexer/sdk/model/WorkflowRun.java
package io.fivexer.sdk.model;

import java.util.List;
import java.util.Map;

/**
 * A running (or finished) instance of a workflow definition.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkflowRun {
    private String id;
    private String workflowId;
    private String status;
    private String currentStepId;
    private String currentTaskId;
    private String initiatorWorkerId;
    private Map<String, Object> context;
    private int definitionVersion;
    private List<WorkflowRunHistoryEntry> history;
    private List<ParallelBranch> parallelBranches;
    private long createdAt;
    private long updatedAt;

    public String getId() { return id; }
    public String getWorkflowId() { return workflowId; }
    public String getStatus() { return status; }
    public String getCurrentStepId() { return currentStepId; }
    public String getCurrentTaskId() { return currentTaskId; }
    public String getInitiatorWorkerId() { return initiatorWorkerId; }
    public Map<String, Object> getContext() { return context; }
    public int getDefinitionVersion() { return definitionVersion; }
    public List<WorkflowRunHistoryEntry> getHistory() { return history; }
    public List<ParallelBranch> getParallelBranches() { return parallelBranches; }
    public long getCreatedAt() { return createdAt; }
    public long getUpdatedAt() { return updatedAt; }
}

File: php/src/Model/WorkflowRun.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * A running (or finished) instance of a workflow definition.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkflowRun
{
    public function __construct(
        public readonly string $id,
        public readonly string $workflowId,
        public readonly string $status,
        public readonly string $initiatorWorkerId,
        public readonly int $definitionVersion,
        public readonly int $createdAt,
        public readonly int $updatedAt,
        public readonly ?string $currentStepId,
        public readonly ?string $currentTaskId,
        /** @var array<string, mixed> */
        public readonly array $context,
        /** @var list<WorkflowRunHistoryEntry> */
        public readonly array $history,
        /** @var list<ParallelBranch>|null */
        public readonly ?array $parallelBranches,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            workflowId: (string) ($data['workflowId'] ?? ''),
            status: (string) ($data['status'] ?? ''),
            initiatorWorkerId: (string) ($data['initiatorWorkerId'] ?? ''),
            definitionVersion: (int) ($data['definitionVersion'] ?? 0),
            createdAt: (int) ($data['createdAt'] ?? 0),
            updatedAt: (int) ($data['updatedAt'] ?? 0),
            currentStepId: isset($data['currentStepId']) ? (string) $data['currentStepId'] : null,
            currentTaskId: isset($data['currentTaskId']) ? (string) $data['currentTaskId'] : null,
            context: $data['context'] ?? [],
            history: Json::parseEach($data, 'history', [WorkflowRunHistoryEntry::class, 'fromArray']),
            parallelBranches: Json::parseEachOrNull($data, 'parallelBranches', [ParallelBranch::class, 'fromArray']),
        );
    }
}
